core/src/main/scala/filodb.core/store/package.scala

```diff
@@ -87,16 +88,37 @@ package object store {
   }
 
   /**
-   * Formulation for chunkID based on a combo of the start time for a chunk and the current time in the lower
-   * bits to disambiguate two chunks which have the same start time.
+   * Formulation for chunkID based on a combo of the start time for a chunk and the ingestion
+   * time in the lower bits to disambiguate two chunks which have the same start time, and to
+   * help with external downsampling based on when the data was ingested.
    *
-   * bits 63-21 (43 bits):  milliseconds since Unix Epoch (1/1/1970) - enough for 278.7 years or through 2248
-   * bits 20-0  (21 bits):  The lower 21 bits of nanotime for disambiguation
+   * bits 63-22 (42 bits): user data start time, as milliseconds since Unix Epoch (1/1/1970) -
+   * enough for 139 years or through the year 2109
+   * bits 21-0 (22 bits): ingestion time, as seconds since Unix Epoch, modulo 48 days to ensure
+   * no wraparound in the middle of the day (full binary encoding supports 48.545 days)
+   *
+   * @param startTime milliseconds since 1970
+   * @param ingestionTime seconds since 1970
    */
-  @inline final def chunkID(startTime: Long): Long = chunkID(startTime, System.nanoTime)
+  @inline final def chunkID(startTime: Long, ingestionTime: Long): Long =
+    (startTime << startTimeShift) | Math.floorMod(ingestionTime, (48 * 24 * 60 * 60))
 
-  @inline final def chunkID(startTime: Long, currentTime: Long): Long =
-    (startTime << msBitOffset) | (currentTime & lowerBitsMask)
+  /**
+   * @param startTime milliseconds since 1970
+   * @param ingestionTime seconds since 1970
+   */
+  @inline final def chunkID(startTime: Instant, ingestionTime: Instant): Long =
+    chunkID(startTime.toEpochMilli(), ingestionTime.getEpochSecond())
+
+  /**
+   * Returns the start time portion of the chunk ID, as milliseconds from 1970.
+   */
+  @inline final def startTimeFromChunkID(chunkID: Long): Long = chunkID >>> startTimeShift
+
+  /**
+   * Returns the ingestion time portion of the chunk ID, as seconds from 1970, modulo 48 days.
+   */
+  @inline final def ingestionTimeFromChunkID(chunkID: Long): Long = chunkID & ingestionTimeMask
```

Comment:
This is not the ingestion Time, it is the ingestionTimeMod48Days.... the method name is kinda misleading